Documentation/cli/starlark.md

```diff
@@ -0,0 +1,265 @@
+recorded() | Equivalent to API call [Recorded](https://godoc.org/github.com/go-delve/delve/service/rpc2#RPCServer.Recorded)
+restart(Position, ResetArgs, NewArgs) | Equivalent to API call [Restart](https://godoc.org/github.com/go-delve/delve/service/rpc2#RPCServer.Restart)
+set_expr(Scope, Symbol, Value) | Equivalent to API call [Set](https://godoc.org/github.com/go-delve/delve/service/rpc2#RPCServer.Set)
+stacktrace(Id, Depth, Full, Defers, Cfg) | Equivalent to API call [Stacktrace](https://godoc.org/github.com/go-delve/delve/service/rpc2#RPCServer.Stacktrace)
+state(NonBlocking) | Equivalent to API call [State](https://godoc.org/github.com/go-delve/delve/service/rpc2#RPCServer.State)
+dlv_command(command) | Executes the specified command as if typed at the dlv_prompt
+read_file(path) | Reads the file as a string
+write_file(path, contents) | Writes string to a file
+cur_scope() | Returns the current evaluation scope
+default_load_config() | Returns the current default load configuration
+<!-- END MAPPING TABLE -->
+
+## Should I use raw_command or dlv_command?
+
+There are two ways to resume the execution of the target program:
+
+	raw_command("continue")
+	dlv_command("continue")
+
+The first one maps to the API call [Command](https://godoc.org/github.com/derekparker/delve/service/rpc2#RPCServer.Command). As such all the caveats explained in the [Client HowTo](../api/ClientHowto.md).
+
+The latter is equivalent to typing `continue` to the `(dlv)` command line and should do what you expect.
+
+In general `dlv_command("continue")` should be preferred, unless the behavior you wish to produces diverges significantly from that of the command line's `continue`.
+
+# Creating new commands
+
+Any global function with a name starting with `command_` will be made available as a command line command. If the function has a single argument named `args` all arguments passed on the command line will be passed to the function as a single string. 
+
+Otherwise arguments passed on the command line are interpreted as starlark expressions. See the [expression arguments](#expression-arguments) example. 
+
+If the command function has a doc string it will be used as a help message.
+
+# Working with variables
+
+Variables of the target program can be accessed using `local_vars`, `function_args` or the `eval` functions. Each variable will be returned as a [Variable](https://godoc.org/github.com/go-delve/delve/service/api#Variable) struct, with one special field: `Value`.
+
+## Variable.Value
+
+The `Value` field will return the value of the target variable converted to a starlark value:
+
+* integers, floating point numbers and strings are represented by equivalent starlark values
+* structs are represented as starlark dictionaries
+* slices and arrays are represented by starlark lists
+* maps are represented by starlark lists where even entries are keys and odd entries are values
```

Comment:
Why not convert to a dictionary instead to preserve the same semantics?

Reply:
We can do it, I thought this was good enough.

Reply:
I think it just breaks expectations. If you're using the starlark API and you want to interact with a map, you'd expect to use normal map operations, not necessarily dealing with an array.

Reply:
Ok, I'll implement it.

Reply:
Done.